src/coredump/coredump.c

```diff
@@ -774,15 +782,44 @@ static int change_uid_gid(const Context *context) {
         return drop_privileges(uid, gid, 0);
 }
 
+static int setup_container_mount_tree(int mount_tree_fd, char **container_root) {
+        _cleanup_free_ char *root = NULL;
+        int r;
+
+        assert(mount_tree_fd >= 0);
+        assert(container_root);
+
+        r = unshare(CLONE_NEWNS);
+        if (r < 0)
+                return log_warning_errno(errno, "Failed to unshare mount namespace: %m");
+
+        r = mount(NULL, "/", NULL, MS_REC|MS_PRIVATE, NULL);
```

Comment:
please do not use MS_PRIVATE if you actually keep a copy of the host mount namespace around. Because that means that removable media and such are not unmounted in this namespace when the main namespace says so. use MS_SHARED